docs/blog/posts/introduction-to-bpfd.md

```diff
@@ -0,0 +1,605 @@
+
+Beyond the fundamentals of developing and loading programs locally to test them,
+once you have working code the process of deploying these in testing and
+staging environments to validate them often ends up resulting in a lot of
+custom program loading and management functionality specific to the application.
+Not to mention eventually moving to production systems, potentially in
+environments like Kubernetes clusters: the operational considerations continue to
+compound.
+
+Security is another significant challenge, which we will cover in more depth in
+a follow-on blog.  However, at a high level, applications that use eBPF
+typically load their own eBPF programs, which requires at least CAP_BPF.  Many
+BPF programs and attach points require additional capabilities from
+CAP_SYS_PTRACE, CAP_NET_ADMIN and even including CAP_SYS_ADMIN.  These
+privileges include capabilities that aren’t strictly necessary for eBPF and are
+too coarsely grained to be useful.  Since the pods that load eBPF are mostly
+long-lived and often don’t drop privileges it leaves a wide attack surface.
+
+While it doesn't solve all the ergonomic and maintenance problems associated
+with adopting eBPF, [bpfd] does try to address several of these issues,
+particularly when it comes to security, and deploying and managing the lifecycle
+of your eBPF programs. In the coming sections, we will go into more depth about
+what eBPF does (and does NOT) do, and how it can help reduce the costs
+associated with deploying and managing eBPF-powered workloads.
+
+[eBPF verifier]:https://docs.kernel.org/bpf/verifier.html
+[bpfd]:https://bpfd.dev
+
+## bpfd Overview
+
+The bpfd project provides a software stack that aims to make it easy to manage
+the full lifecycle of eBPF programs.  In particular, it can load, unload,
+modify, and monitor eBPF programs on a single host, or across a full Kubernetes
+cluster. The key components of the bpfd solution include the bpfd daemon, and
+Kubernetes support composed of eBPF CRDs, a bpfd agent, and a bpfd operator that
+manages the installation and lifecycle of the bpfd agent and CRDs in a
+Kubernetes cluster. bpfd can be used independently on an individual Linux
+system, or with Kubernetes to manage eBPF programs across the entire cluster.
+
+These components will be covered in more detail in the following sections.
+
+### bpfd Daemon
+
+The bpfd daemon works directly with the operating system to manage eBPF
+programs.  It loads, updates, and unloads eBPF programs, pins maps, and provides
+visibility into the eBPF programs loaded on a system.  As of the writing of this
+blog, bpfd fully supports XDP, TC, Tracepoint, uProbe, and kProbe eBPF programs.
+In addition, bpfd can display information about all types of eBPF programs
+loaded on a system whether they were loaded by bpfd or some other mechanism.
+bpfd is developed in the Rust programming language and uses [Aya], an eBPF
+library which is also developed in Rust.
```

Comment:
Can we add a quick note here on WHY it was developed in rust?

Reply:
I guess it's for all the reasons that people say rust is good.  However, in the spirit of not making this longer than it already is, I'm inclined to leave that out.